### PR TITLE
berkeley-db: Correct TLS auto-detection

### LINKED ISSUE #32591: Installation issue: berkeley-db
### Steps to reproduce the issue

```console
spack install berkeley-db %clang @15
```

### Error message

A (Spack-built) Clang 15 cannot build berkeley-db. The problem is that berkeley-db's configure script checks whether TLS is available. This configure check has a bug, as it uses `NULL` without including `<stdlib.h>`. Thus the configure check always fails event though TLS is available, and the build later errors out since TLS is always required.

### Information on your system

Not relevant; I already have a patch.

### Additional information

Not relevant; I already have a patch.

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

### PR DESCRIPTION
**Pull request by eschnett**
Closes https://github.com/spack/spack/issues/32591.

### CODE REVIEW

File: var/spack/repos/builtin/packages/berkeley-db/package.py [hunk 1]
```diff
@@ -39,6 +39,7 @@ class BerkeleyDb(AutotoolsPackage):
     build_directory = "build_unix"
 
     patch("drop-docs.patch", when="~docs")
+    patch("tls.patch")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/berkeley-db/package.py:46]: Can you add a comment above the `patch` directive with provenance, aim of the patch etc. ?

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/berkeley-db/package.py (hunk 1 above)
Comment (eschnett) [reply var/spack/repos/builtin/packages/berkeley-db/package.py:46]: Done.

**Review by alalazo — APPROVED** [review]

**Comment by alalazo** [comment]
@spackbot run pipeline

File: var/spack/repos/builtin/packages/berkeley-db/package.py (hunk 1 above)
Comment (eschnett) [reply var/spack/repos/builtin/packages/berkeley-db/package.py:46]: Done.

**Pull request merged**